{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Ignore warnings\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting biopython\n",
      "  Downloading biopython-1.85-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)\n",
      "Requirement already satisfied: numpy in /home/leandro/Insync/leandro.radusky@gmail.com/Google Drive/Mimark/code/cohort-analysis/.venv/lib/python3.13/site-packages (from biopython) (2.2.3)\n",
      "Downloading biopython-1.85-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.3 MB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.3/3.3 MB\u001b[0m \u001b[31m1.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: biopython\n",
      "Successfully installed biopython-1.85\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "# %pip install biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import the sbg workspace and local utils\n",
    "import sys\n",
    "sys.path.append('/home/leandro/Dropbox/workspacesbg/sbg/')\n",
    "sys.path.append('../../../code/')\n",
    "\n",
    "from sbg.orf.Orf import OnlineOrf\n",
    "from sbg.structure.AlphaFoldHandler import AlphaFoldHandler\n",
    "from sbg.structure.Structure import Structure\n",
    "from sbg.common.FileHandler import FileHandler\n",
    "from sbg.scripts.foldx.TangoHandler import TangoHandler\n",
    "\n",
    "import pipeline_functions as pf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'Bio'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 4\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mpandas\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mas\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mpd\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mos\u001b[39;00m\n\u001b[0;32m----> 4\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01msequence_utils\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m highlight_sequence, align_sequences_biopython, display_alignment_with_highlighting\n",
      "File \u001b[0;32m~/Insync/leandro.radusky@gmail.com/Google Drive/Mimark/code/cohort-analysis/notebooks/other_analyses/antigens/../../../code/sequence_utils.py:1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mBio\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m AlignIO\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mBio\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mAlign\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m MultipleSeqAlignment\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mBio\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mSeqRecord\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m SeqRecord\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'Bio'"
     ]
    }
   ],
   "source": [
    "# External libraries\n",
    "import pandas as pd\n",
    "import os\n",
    "from sequence_utils import highlight_sequence, align_sequences_biopython, display_alignment_with_highlighting\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from data/protein_data.csv\n",
    "# data = pd.read_csv('data/protein_data.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Run for Irene\n",
    "up_accs = ['P05067','P02452','O00468','Q5S248','P10451','P35321','A8K2U0','P22528','P50454','P04275','P16035','P80188']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computation Pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing P07355 ANXA2_HUMAN, 1 of 505\n",
      "Processing O00425 IF2B3_HUMAN, 2 of 505\n",
      "Processing O15392 BIRC5_HUMAN, 3 of 505\n",
      "Processing P00749 UROK_HUMAN, 4 of 505\n",
      "Processing P02787 TRFE_HUMAN, 5 of 505\n",
      "Processing P03372 ESR1_HUMAN, 6 of 505\n",
      "Processing P04792 HSPB1_HUMAN, 7 of 505\n",
      "Processing P07339 CATD_HUMAN, 8 of 505\n",
      "Processing P08253 MMP2_HUMAN, 9 of 505\n",
      "Processing P09603 CSF1_HUMAN, 10 of 505\n",
      "Processing P10415 BCL2_HUMAN, 11 of 505\n",
      "Processing P11166 GTR1_HUMAN, 12 of 505\n",
      "Processing P14780 MMP9_HUMAN, 13 of 505\n",
      "Processing P15692 VEGFA_HUMAN, 14 of 505\n",
      "Processing P15941 MUC1_HUMAN, 15 of 505\n",
      "Processing P17813 EGLN_HUMAN, 16 of 505\n",
      "Processing P35354 PGH2_HUMAN, 17 of 505\n",
      "Processing P38936 CDN1A_HUMAN, 18 of 505\n",
      "Processing P42771 CD2A1_HUMAN, 19 of 505\n",
      "Processing P60484 PTEN_HUMAN, 20 of 505\n",
      "Processing P61604 CH10_HUMAN, 21 of 505\n",
      "Processing Q13938 CAYP1_HUMAN, 22 of 505\n",
      "Processing Q14508 WFDC2_HUMAN, 23 of 505\n",
      "Processing P12830 CADH1_HUMAN, 24 of 505\n",
      "Processing P11802 CDK4_HUMAN, 25 of 505\n",
      "Processing P17661 DESM_HUMAN, 26 of 505\n",
      "Processing O14497 ARI1A_HUMAN, 27 of 505\n",
      "Processing Q07812 BAX_HUMAN, 28 of 505\n",
      "Processing P04626 ERBB2_HUMAN, 29 of 505\n",
      "Processing P04637 P53_HUMAN, 30 of 505\n",
      "Processing Q8WXI7 MUC16_HUMAN, 31 of 505\n",
      "Processing P46013 KI67_HUMAN, 32 of 505\n",
      "Processing Q9Y251 HPSE_HUMAN, 33 of 505\n",
      "Processing P35221 CTNA1_HUMAN, 34 of 505\n",
      "Processing Q15118 PDK1_HUMAN, 35 of 505\n",
      "Processing P43405 KSYK_HUMAN, 36 of 505\n",
      "Processing Q92878 RAD50_HUMAN, 37 of 505\n",
      "Processing O96017 CHK2_HUMAN, 38 of 505\n",
      "Processing P24864 uniprot_accession\n",
      "P24864    CCNE1_HUMAN\n",
      "P24864    CCNE1_HUMAN\n",
      "Name: swiss_prot, dtype: object, 39 of 505\n",
      "Processing P06493 CDK1_HUMAN, 40 of 505\n",
      "Processing P08243 ASNS_HUMAN, 41 of 505\n",
      "Processing P15328 FOLR1_HUMAN, 42 of 505\n",
      "Processing P40692 MLH1_HUMAN, 43 of 505\n",
      "Processing P14618 KPY1_HUMAN, 44 of 505\n",
      "Processing P07237 PDIA1_HUMAN, 45 of 505\n",
      "Processing P06401 PRGR_HUMAN, 46 of 505\n",
      "Processing P19544 WT1_HUMAN, 47 of 505\n",
      "Processing O14493 CLD4_HUMAN, 48 of 505\n",
      "Processing P16035 TIMP2_HUMAN, 49 of 505\n",
      "Processing P35222 CTNB1_HUMAN, 50 of 505\n",
      "Processing P48023 TNFL6_HUMAN, 51 of 505\n",
      "Processing A2N4P8 A2N4P8_HUMAN, 52 of 505\n",
      "Gene 'IL2R' not found in the expression data.\n",
      "Processing O95997 PTTG1_HUMAN, 53 of 505\n",
      "Processing P00533 EGFR_HUMAN, 54 of 505\n",
      "Processing P01106 MYC_HUMAN, 55 of 505\n",
      "Processing P01236 PRL_HUMAN, 56 of 505\n",
      "Processing P01241 SOMA_HUMAN, 57 of 505\n",
      "Processing P02647 APOA1_HUMAN, 58 of 505\n",
      "Processing P02741 CRP_HUMAN, 59 of 505\n",
      "Processing P02766 TTHY_HUMAN, 60 of 505\n",
      "Processing P02771 FETA_HUMAN, 61 of 505\n",
      "Processing P04083 ANXA1_HUMAN, 62 of 505\n",
      "Processing P08254 MMP3_HUMAN, 63 of 505\n",
      "Processing P08476 INHBA_HUMAN, 64 of 505\n",
      "Processing P0DJI8 SAA1_HUMAN, 65 of 505\n",
      "Processing P10451 OSTP_HUMAN, 66 of 505\n",
      "Processing P11021 GRP78_HUMAN, 67 of 505\n",
      "Processing P30040 ERP29_HUMAN, 68 of 505\n",
      "Processing P13647 K2C5_HUMAN, 69 of 505\n",
      "Processing P16070 CD44_HUMAN, 70 of 505\n",
      "Processing P16949 STMN1_HUMAN, 71 of 505\n",
      "Processing P17931 LEG3_HUMAN, 72 of 505\n",
      "Processing P17948 VGFR1_HUMAN, 73 of 505\n",
      "Processing P28906 CD34_HUMAN, 74 of 505\n",
      "Processing P29317 EPHA2_HUMAN, 75 of 505\n",
      "Processing P29508 SPB3_HUMAN, 76 of 505\n",
      "Processing P30086 PEBP1_HUMAN, 77 of 505\n",
      "Processing P35638 DDIT3_HUMAN, 78 of 505\n",
      "Processing P37275 ZEB1_HUMAN, 79 of 505\n",
      "Processing P42345 MTOR_HUMAN, 80 of 505\n",
      "Processing P43246 MSH2_HUMAN, 81 of 505\n",
      "Processing Q01469 FABP5_HUMAN, 82 of 505\n",
      "Processing Q15672 TWST1_HUMAN, 83 of 505\n",
      "Processing Q15848 ADIPO_HUMAN, 84 of 505\n",
      "Processing Q16665 HIF1A_HUMAN, 85 of 505\n",
      "Processing P21802 FGFR2_HUMAN, 86 of 505\n",
      "Processing Q01995 TAGL_HUMAN, 87 of 505\n",
      "Processing P46527 CDN1B_HUMAN, 88 of 505\n",
      "Processing P14635 CCNB1_HUMAN, 89 of 505\n",
      "Processing P31749 AKT1_HUMAN, 90 of 505\n",
      "Processing P40763 STAT3_HUMAN, 91 of 505\n",
      "Processing P01833 PIGR_HUMAN, 92 of 505\n",
      "Processing Q969H0 FBXW7_HUMAN, 93 of 505\n",
      "Processing P30153 2AAA_HUMAN, 94 of 505\n",
      "Processing Q07864 DPOE1_HUMAN, 95 of 505\n",
      "Processing P27986 P85A_HUMAN, 96 of 505\n",
      "Processing P35268 RL22_HUMAN, 97 of 505\n",
      "Processing Q14865 ARI5B_HUMAN, 98 of 505\n",
      "Processing P09237 MMP7_HUMAN, 99 of 505\n",
      "Processing P21583 SCF_HUMAN, 100 of 505\n",
      "Processing P11388 TOP2A_HUMAN, 101 of 505\n",
      "Processing P60174 TPIS_HUMAN, 102 of 505\n",
      "Processing P05109 S10A8_HUMAN, 103 of 505\n",
      "Processing Q13635 PTC1_HUMAN, 104 of 505\n",
      "Processing Q15465 SHH_HUMAN, 105 of 505\n",
      "Processing Q99835 SMO_HUMAN, 106 of 505\n",
      "Processing P02461 CO3A1_HUMAN, 107 of 505\n",
      "Processing P01233 CGHB_HUMAN, 108 of 505\n",
      "Error with P01233 in step 1: type object 'new' has no attribute 'entry'\n",
      "Processing O75390 CISY_HUMAN, 109 of 505\n",
      "Processing P05787 K2C8_HUMAN, 110 of 505\n",
      "Processing P09493 TPM1_HUMAN, 111 of 505\n",
      "Processing P22626 ROA2_HUMAN, 112 of 505\n",
      "Processing O00559 RCAS1_HUMAN, 113 of 505\n",
      "Processing O76070 SYUG_HUMAN, 114 of 505\n",
      "Processing P01033 TIMP1_HUMAN, 115 of 505\n",
      "Processing P02538 K2C6A_HUMAN, 116 of 505\n",
      "Processing P04259 K2C6B_HUMAN, 117 of 505\n",
      "Processing P05111 INHA_HUMAN, 118 of 505\n",
      "Processing P05121 PAI1_HUMAN, 119 of 505\n",
      "Processing P09211 uniprot_accession\n",
      "P09211    GSTP1_HUMAN\n",
      "P09211    GSTP1_HUMAN\n",
      "Name: swiss_prot, dtype: object, 120 of 505\n",
      "Processing P14373 TRI27_HUMAN, 121 of 505\n",
      "Processing P19440 GGT1_HUMAN, 122 of 505\n",
      "Processing P22223 CADH3_HUMAN, 123 of 505\n",
      "Processing Q9H3D4 P63_HUMAN, 124 of 505\n",
      "Processing P01009 A1AT_HUMAN, 125 of 505\n",
      "Processing P62937 PPIA_HUMAN, 126 of 505\n",
      "Processing P42336 PK3CA_HUMAN, 127 of 505\n",
      "Processing O54908 DKK1_MOUSE, 128 of 505\n",
      "Gene 'Dkk1' not found in the expression data.\n",
      "Processing P00441 SODC_HUMAN, 129 of 505\n",
      "Processing P01137 TGFB1_HUMAN, 130 of 505\n",
      "Processing P05019 IGF1_HUMAN, 131 of 505\n",
      "Processing P07333 CSF1R_HUMAN, 132 of 505\n",
      "Processing P07858 CATB_HUMAN, 133 of 505\n",
      "Processing P08670 VIME_HUMAN, 134 of 505\n",
      "Processing P09038 FGF2_HUMAN, 135 of 505\n",
      "Processing P0C0L4 CO4A_HUMAN, 136 of 505\n",
      "Processing P24385 CCND1_HUMAN, 137 of 505\n",
      "Processing P35968 VGFR2_HUMAN, 138 of 505\n",
      "Processing P42574 CASP3_HUMAN, 139 of 505\n",
      "Processing P46531 NOTC1_HUMAN, 140 of 505\n",
      "Processing P54760 EPHB4_HUMAN, 141 of 505\n",
      "Processing Q02962 PAX2_HUMAN, 142 of 505\n",
      "Processing Q14624 ITIH4_HUMAN, 143 of 505\n",
      "Processing Q16658 FSCN1_HUMAN, 144 of 505\n",
      "Processing Q99572 P2RX7_HUMAN, 145 of 505\n",
      "Processing Q9UBP4 DKK3_HUMAN, 146 of 505\n",
      "Processing P52926 HMGA2_HUMAN, 147 of 505\n",
      "Processing Q13753 LAMC2_HUMAN, 148 of 505\n",
      "Processing O00755 WNT7A_HUMAN, 149 of 505\n",
      "Processing O43597 SPY2_HUMAN, 150 of 505\n",
      "Processing O96028 nan, 151 of 505\n",
      "Gene 'NSD2' not found in the expression data.\n",
      "Processing P01135 TGFA_HUMAN, 152 of 505\n",
      "Processing P05164 PERM_HUMAN, 153 of 505\n",
      "Processing P05231 IL6_HUMAN, 154 of 505\n",
      "Processing P05362 ICAM1_HUMAN, 155 of 505\n",
      "Processing Q06830 PRDX1_HUMAN, 156 of 505\n",
      "Processing P08833 IBP1_HUMAN, 157 of 505\n",
      "Processing P10147 CCL3_HUMAN, 158 of 505\n",
      "Processing P13236 CCL4_HUMAN, 159 of 505\n",
      "Processing P13501 CCL5_HUMAN, 160 of 505\n",
      "Processing P16581 LYAM2_HUMAN, 161 of 505\n",
      "Processing P19320 VCAM1_HUMAN, 162 of 505\n",
      "Processing P19438 TNR1A_HUMAN, 163 of 505\n",
      "Processing P21741 MK_HUMAN, 164 of 505\n",
      "Processing P22894 MMP8_HUMAN, 165 of 505\n",
      "Processing P29965 CD40L_HUMAN, 166 of 505\n",
      "Processing P33993 MCM7_HUMAN, 167 of 505\n",
      "Processing P43490 NAMPT_HUMAN, 168 of 505\n",
      "Processing P49789 FHIT_HUMAN, 169 of 505\n",
      "Processing P51671 CCL11_HUMAN, 170 of 505\n",
      "Processing P80188 NGAL_HUMAN, 171 of 505\n",
      "Processing Q04743 EMX2_HUMAN, 172 of 505\n",
      "Processing Q13421 MSLN_HUMAN, 173 of 505\n",
      "Processing Q16629 SRSF7_HUMAN, 174 of 505\n",
      "Processing Q29983 MICA_HUMAN, 175 of 505\n",
      "Processing Q99988 GDF15_HUMAN, 176 of 505\n",
      "Processing Q9BZM4 N2DL3_HUMAN, 177 of 505\n",
      "Processing Q9BZM6 N2DL1_HUMAN, 178 of 505\n",
      "Processing Q9Y6M1 IF2B2_HUMAN, 179 of 505\n",
      "Processing Q15831 STK11_HUMAN, 180 of 505\n",
      "Processing P06396 GELS_HUMAN, 181 of 505\n",
      "Processing P06733 ENOA_HUMAN, 182 of 505\n",
      "Processing P07951 TPM2_HUMAN, 183 of 505\n",
      "Processing P12109 CO6A1_HUMAN, 184 of 505\n",
      "Processing P21980 TGM2_HUMAN, 185 of 505\n",
      "Processing P24844 MYL9_HUMAN, 186 of 505\n",
      "Processing P27338 AOFB_HUMAN, 187 of 505\n",
      "Processing P51911 CNN1_HUMAN, 188 of 505\n",
      "Processing P62736 ACTA_HUMAN, 189 of 505\n",
      "Processing P63261 ACTG_HUMAN, 190 of 505\n",
      "Processing Q15124 PGM5_HUMAN, 191 of 505\n",
      "Processing Q5TCI8 Q5TCI8_HUMAN, 192 of 505\n",
      "Processing Q93052 LPP_HUMAN, 193 of 505\n",
      "Processing P50225 ST1A1_HUMAN, 194 of 505\n",
      "Processing Q92731 ESR2_HUMAN, 195 of 505\n",
      "Processing O43683 BUB1_HUMAN, 196 of 505\n",
      "Processing Q16851 UGPA_HUMAN, 197 of 505\n",
      "Processing P41161 ETV5_HUMAN, 198 of 505\n",
      "Processing Q04609 FOLH1_HUMAN, 199 of 505\n",
      "Processing Q16512 PKN1_HUMAN, 200 of 505\n",
      "Processing Q13043 STK4_HUMAN, 201 of 505\n",
      "Processing Q92597 NDRG1_HUMAN, 202 of 505\n",
      "Processing O15120 PLCB_HUMAN, 203 of 505\n",
      "Processing O15551 CLD3_HUMAN, 204 of 505\n",
      "Processing P21810 PGS1_HUMAN, 205 of 505\n",
      "Processing P06731 CEAM5_HUMAN, 206 of 505\n",
      "Processing P26447 S10A4_HUMAN, 207 of 505\n",
      "Processing Q13309 SKP2_HUMAN, 208 of 505\n",
      "Processing P05412 JUN_HUMAN, 209 of 505\n",
      "Processing P52799 EFNB2_HUMAN, 210 of 505\n",
      "Processing P09529 INHBB_HUMAN, 211 of 505\n",
      "Processing P09758 TACD2_HUMAN, 212 of 505\n",
      "Processing P11511 CP19A_HUMAN, 213 of 505\n",
      "Processing P19971 TYPH_HUMAN, 214 of 505\n",
      "Processing P31947 1433S_HUMAN, 215 of 505\n",
      "Processing P48509 CD151_HUMAN, 216 of 505\n",
      "Processing Q14457 BECN1_HUMAN, 217 of 505\n",
      "Processing O95994 AGR2_HUMAN, 218 of 505\n",
      "Processing P00338 LDHA_HUMAN, 219 of 505\n",
      "Processing P01023 A2MG_HUMAN, 220 of 505\n",
      "Processing P01375 TNFA_HUMAN, 221 of 505\n",
      "Processing P02768 ALBU_HUMAN, 222 of 505\n",
      "Processing P04179 SODM_HUMAN, 223 of 505\n",
      "Processing P04217 A1BG_HUMAN, 224 of 505\n",
      "Processing P09382 LEG1_HUMAN, 225 of 505\n",
      "Processing P10909 CLUS_HUMAN, 226 of 505\n",
      "Processing P12004 PCNA_HUMAN, 227 of 505\n",
      "Processing P12277 KCRB_HUMAN, 228 of 505\n",
      "Processing P25445 TNR6_HUMAN, 229 of 505\n",
      "Processing P32004 L1CAM_HUMAN, 230 of 505\n",
      "Processing P40121 CAPG_HUMAN, 231 of 505\n",
      "Processing P50454 nan, 232 of 505\n",
      "Processing P08758 nan, 233 of 505\n",
      "Processing P16234 PGFRA_HUMAN, 234 of 505\n",
      "Processing P35900 K1C20_HUMAN, 235 of 505\n",
      "Processing Q03405 UPAR_HUMAN, 236 of 505\n",
      "Processing P01111 RASN_HUMAN, 237 of 505\n",
      "Processing O00299 CLIC1_HUMAN, 238 of 505\n",
      "Processing O14746 TERT_HUMAN, 239 of 505\n",
      "Processing O14965 AURKA_HUMAN, 240 of 505\n",
      "Processing O15123 ANGP2_HUMAN, 241 of 505\n",
      "Processing O60271 JIP4_HUMAN, 242 of 505\n",
      "Processing O75556 SG2A1_HUMAN, 243 of 505\n",
      "Processing P00352 nan, 244 of 505\n",
      "Processing P01024 CO3_HUMAN, 245 of 505\n",
      "Processing P01344 IGF2_HUMAN, 246 of 505\n",
      "Processing P02654 APOC1_HUMAN, 247 of 505\n",
      "Processing P05090 APOD_HUMAN, 248 of 505\n",
      "Processing P06400 RB_HUMAN, 249 of 505\n",
      "Processing P06727 APOA4_HUMAN, 250 of 505\n",
      "Processing P30044 PRDX5_HUMAN, 251 of 505\n",
      "Processing P30041 PRDX6_HUMAN, 252 of 505\n",
      "Processing Q99623 PHB2_HUMAN, 253 of 505\n",
      "Processing P08069 IGF1R_HUMAN, 254 of 505\n",
      "Processing P08151 GLI1_HUMAN, 255 of 505\n",
      "Processing P10070 GLI2_HUMAN, 256 of 505\n",
      "Processing P10071 GLI3_HUMAN, 257 of 505\n",
      "Processing P17936 uniprot_accession\n",
      "P17936    IBP3_HUMAN\n",
      "P17936    IBP3_HUMAN\n",
      "Name: swiss_prot, dtype: object, 258 of 505\n",
      "Processing P25063 CD24_HUMAN, 259 of 505\n",
      "Processing P26006 ITA3_HUMAN, 260 of 505\n",
      "Processing P33527 MRP1_HUMAN, 261 of 505\n",
      "Processing P35916 VGFR3_HUMAN, 262 of 505\n",
      "Processing P41159 LEP_HUMAN, 263 of 505\n",
      "Processing P49767 VEGFC_HUMAN, 264 of 505\n",
      "Processing P55060 XPO2_HUMAN, 265 of 505\n",
      "Processing P67870 CSK2B_HUMAN, 266 of 505\n",
      "Processing P78396 CCNA1_HUMAN, 267 of 505\n",
      "Processing P78504 JAG1_HUMAN, 268 of 505\n",
      "Processing Q02763 TIE2_HUMAN, 269 of 505\n",
      "Processing Q07654 TFF3_HUMAN, 270 of 505\n",
      "Processing Q12983 BNIP3_HUMAN, 271 of 505\n",
      "Processing Q15506 SP17_HUMAN, 272 of 505\n",
      "Processing Q8N307 MUC20_HUMAN, 273 of 505\n",
      "Processing Q96I34 PP16A_HUMAN, 274 of 505\n",
      "Processing Q9H013 ADA19_HUMAN, 275 of 505\n",
      "Processing Q9H6S3 ES8L2_HUMAN, 276 of 505\n",
      "Processing Q9NR61 DLL4_HUMAN, 277 of 505\n",
      "Processing Q9NY33 DPP3_HUMAN, 278 of 505\n",
      "Processing Q9UM47 NOTC3_HUMAN, 279 of 505\n",
      "Processing Q9UMX1 SUFU_HUMAN, 280 of 505\n",
      "Processing Q9Y6Q9 NCOA3_HUMAN, 281 of 505\n",
      "Processing O60216 RAD21_HUMAN, 282 of 505\n",
      "Processing Q01196 RUNX1_HUMAN, 283 of 505\n",
      "Processing Q9Y6Q5 AP1M2_HUMAN, 284 of 505\n",
      "Processing P01116 RASK_HUMAN, 285 of 505\n",
      "Processing P22309 UD11_HUMAN, 286 of 505\n",
      "Processing P48061 SDF1_HUMAN, 287 of 505\n",
      "Processing P23219 PGH1_HUMAN, 288 of 505\n",
      "Processing Q15910 EZH2_HUMAN, 289 of 505\n",
      "Processing P09466 PAEP_HUMAN, 290 of 505\n",
      "Processing Q14103 ROD_HUMAN, 291 of 505\n",
      "Processing P00750 TPA_HUMAN, 292 of 505\n",
      "Processing P14174 MIF_HUMAN, 293 of 505\n",
      "Processing Q15389 ANGP1_HUMAN, 294 of 505\n",
      "Processing P21926 CD9_HUMAN, 295 of 505\n",
      "Processing P39060 COIA1_HUMAN, 296 of 505\n",
      "Processing P35625 TIMP3_HUMAN, 297 of 505\n",
      "Processing P21964 COMT_HUMAN, 298 of 505\n",
      "Processing Q99814 EPAS1_HUMAN, 299 of 505\n",
      "Processing O43278 SPIT1_HUMAN, 300 of 505\n",
      "Processing O43291 SPIT2_HUMAN, 301 of 505\n",
      "Processing P17936 uniprot_accession\n",
      "P17936    IBP3_HUMAN\n",
      "P17936    IBP3_HUMAN\n",
      "Name: swiss_prot, dtype: object, 302 of 505\n",
      "Processing P10914 IRF1_HUMAN, 303 of 505\n",
      "Processing Q6XLA1 CSC2A_HUMAN, 304 of 505\n",
      "Processing O94788 AL1A2_HUMAN, 305 of 505\n",
      "Processing Q8NBS9 nan, 306 of 505\n",
      "Processing Q86YT6 MIB1_HUMAN, 307 of 505\n",
      "Processing O43490 PROM1_HUMAN, 308 of 505\n",
      "Processing P09455 RET1_HUMAN, 309 of 505\n",
      "Processing P14902 I23O1_HUMAN, 310 of 505\n",
      "Processing P25116 PAR1_HUMAN, 311 of 505\n",
      "Processing P27658 CO8A1_HUMAN, 312 of 505\n",
      "Processing P35670 ATP7B_HUMAN, 313 of 505\n",
      "Processing P49327 FAS_HUMAN, 314 of 505\n",
      "Processing P52701 MSH6_HUMAN, 315 of 505\n",
      "Processing Q96GD4 AURKB_HUMAN, 316 of 505\n",
      "Processing Q9BY76 ANGL4_HUMAN, 317 of 505\n",
      "Processing Q9UIQ6 LCAP_HUMAN, 318 of 505\n",
      "Processing Q9HC84 MUC5B_HUMAN, 319 of 505\n",
      "Processing P21860 ERBB3_HUMAN, 320 of 505\n",
      "Processing O00762 UBE2C_HUMAN, 321 of 505\n",
      "Processing O60259 KLK8_HUMAN, 322 of 505\n",
      "Processing O95832 CLD1_HUMAN, 323 of 505\n",
      "Processing P00450 CERU_HUMAN, 324 of 505\n",
      "Processing P00738 nan, 325 of 505\n",
      "Processing P01008 ANT3_HUMAN, 326 of 505\n",
      "Processing P02452 CO1A1_HUMAN, 327 of 505\n",
      "Processing P02652 APOA2_HUMAN, 328 of 505\n",
      "Processing P02671 nan, 329 of 505\n",
      "Processing P02675 nan, 330 of 505\n",
      "Processing P02749 APOH_HUMAN, 331 of 505\n",
      "Processing P02765 FETUA_HUMAN, 332 of 505\n",
      "Processing P02774 nan, 333 of 505\n",
      "Processing P02775 CXCL7_HUMAN, 334 of 505\n",
      "Processing P02776 nan, 335 of 505\n",
      "Processing P02790 nan, 336 of 505\n",
      "Processing P05198 IF2A_HUMAN, 337 of 505\n",
      "Processing P05783 nan, 338 of 505\n",
      "Processing P06703 nan, 339 of 505\n",
      "Processing P07108 ACBP_HUMAN, 340 of 505\n",
      "Processing P08107 HSP71_HUMAN, 341 of 505\n",
      "Error with P08107 in step 1: type object 'new' has no attribute 'entry'\n",
      "Processing P08123 CO1A2_HUMAN, 342 of 505\n",
      "Processing P11678 PERE_HUMAN, 343 of 505\n",
      "Processing P15531 NDKA_HUMAN, 344 of 505\n",
      "Processing P18827 SDC1_HUMAN, 345 of 505\n",
      "Processing P18850 ATF6A_HUMAN, 346 of 505\n",
      "Processing P19338 nan, 347 of 505\n",
      "Processing P19404 NDUV2_HUMAN, 348 of 505\n",
      "Processing P19823 nan, 349 of 505\n",
      "Processing P20809 IL11_HUMAN, 350 of 505\n",
      "Processing P21333 nan, 351 of 505\n",
      "Processing P22392 NDKB_HUMAN, 352 of 505\n",
      "Processing P23142 nan, 353 of 505\n",
      "Processing P23526 SAHH_HUMAN, 354 of 505\n",
      "Processing P24821 TENA_HUMAN, 355 of 505\n",
      "Processing P29966 nan, 356 of 505\n",
      "Processing P31321 KAP1_HUMAN, 357 of 505\n",
      "Processing P31943 HNRH1_HUMAN, 358 of 505\n",
      "Processing P35232 PHB_HUMAN, 359 of 505\n",
      "Gene 'PHB1' not found in the expression data.\n",
      "Processing P36222 CH3L1_HUMAN, 360 of 505\n",
      "Processing P41134 ID1_HUMAN, 361 of 505\n",
      "Processing P42330 AK1C3_HUMAN, 362 of 505\n",
      "Processing P47756 CAPZB_HUMAN, 363 of 505\n",
      "Processing P55072 TERA_HUMAN, 364 of 505\n",
      "Processing P62851 nan, 365 of 505\n",
      "Processing Q01518 CAP1_HUMAN, 366 of 505\n",
      "Processing Q12809 KCNH2_HUMAN, 367 of 505\n",
      "Processing Q12906 nan, 368 of 505\n",
      "Processing Q13162 PRDX4_HUMAN, 369 of 505\n",
      "Processing Q13761 RUNX3_HUMAN, 370 of 505\n",
      "Processing Q14315 nan, 371 of 505\n",
      "Processing Q14626 I11RA_HUMAN, 372 of 505\n",
      "Processing Q15121 nan, 373 of 505\n",
      "Processing Q16555 Q16555, 374 of 505\n",
      "Processing Q32Q65 Q32Q65_HUMAN, 375 of 505\n",
      "Processing Q6FHJ7 nan, 376 of 505\n",
      "Processing Q6NUR7 Q6NUR7_HUMAN, 377 of 505\n",
      "Processing Q99832 TCPH_HUMAN, 378 of 505\n",
      "Processing Q9NSC7 SIA7A_HUMAN, 379 of 505\n",
      "Processing Q9NY64 GTR8_HUMAN, 380 of 505\n",
      "Processing Q9UBU3 GHRL_HUMAN, 381 of 505\n",
      "Processing Q9Y263 PLAP_HUMAN, 382 of 505\n",
      "Processing O14508 SOCS2_HUMAN, 383 of 505\n",
      "Processing O95049 ZO3_HUMAN, 384 of 505\n",
      "Processing O95967 FBLN4_HUMAN, 385 of 505\n",
      "Processing P02751 FINC_HUMAN, 386 of 505\n",
      "Processing P07585 PGS2_HUMAN, 387 of 505\n",
      "Processing P09110 THIK_HUMAN, 388 of 505\n",
      "Processing P0DJI9 SAA2_HUMAN, 389 of 505\n",
      "Processing P10696 PPBN_HUMAN, 390 of 505\n",
      "Gene 'ALPG' not found in the expression data.\n",
      "Processing P15735 PHKG2_HUMAN, 391 of 505\n",
      "Processing P42830 CXCL5_HUMAN, 392 of 505\n",
      "Processing P78556 CCL20_HUMAN, 393 of 505\n",
      "Processing Q02833 RASF7_HUMAN, 394 of 505\n",
      "Processing Q08345 DDR1_HUMAN, 395 of 505\n",
      "Processing Q10589 BST2_HUMAN, 396 of 505\n",
      "Processing Q12756 KIF1A_HUMAN, 397 of 505\n",
      "Processing Q14164 IKKE_HUMAN, 398 of 505\n",
      "Processing Q15915 ZIC1_HUMAN, 399 of 505\n",
      "Processing Q53R41 FAKD1_HUMAN, 400 of 505\n",
      "Processing Q5TEJ8 THMS2_HUMAN, 401 of 505\n",
      "Processing Q8N6T7 SIR6_HUMAN, 402 of 505\n",
      "Processing Q8ND30 LIPB2_HUMAN, 403 of 505\n",
      "Processing Q96D59 RN183_HUMAN, 404 of 505\n",
      "Processing Q99571 P2RX4_HUMAN, 405 of 505\n",
      "Processing Q99784 NOE1_HUMAN, 406 of 505\n",
      "Processing Q9P107 GMIP_HUMAN, 407 of 505\n",
      "Processing Q9P2M7 CING_HUMAN, 408 of 505\n",
      "Processing Q9UBT3 DKK4_HUMAN, 409 of 505\n",
      "Processing P08581 MET_HUMAN, 410 of 505\n",
      "Processing P01112 RASH_HUMAN, 411 of 505\n",
      "Processing P23443 KS6B1_HUMAN, 412 of 505\n",
      "Processing P09769 FGR_HUMAN, 413 of 505\n",
      "Processing P24864 uniprot_accession\n",
      "P24864    CCNE1_HUMAN\n",
      "P24864    CCNE1_HUMAN\n",
      "Name: swiss_prot, dtype: object, 414 of 505\n",
      "Processing P10145 IL8_HUMAN, 415 of 505\n",
      "Processing Q13111 CAF1A_HUMAN, 416 of 505\n",
      "Processing Q13541 4EBP1_HUMAN, 417 of 505\n",
      "Processing P01042 KNG1_HUMAN, 418 of 505\n",
      "Processing P28838 AMPL_HUMAN, 419 of 505\n",
      "Processing P17301 ITA2_HUMAN, 420 of 505\n",
      "Processing O43852 CALU_HUMAN, 421 of 505\n",
      "Processing P02788 TRFL_HUMAN, 422 of 505\n",
      "Processing P19838 NFKB1_HUMAN, 423 of 505\n",
      "Processing P00918 CAH2_HUMAN, 424 of 505\n",
      "Processing Q16790 CAH9_HUMAN, 425 of 505\n",
      "Processing O43570 CAH12_HUMAN, 426 of 505\n",
      "Processing P53350 PLK1_HUMAN, 427 of 505\n",
      "Processing Q86XX4 FRAS1_HUMAN, 428 of 505\n",
      "Processing P09874 PARP1_HUMAN, 429 of 505\n",
      "Processing P31949 S111_HUMAN, 430 of 505\n",
      "Processing Q99102 MUC4_HUMAN, 431 of 505\n",
      "Processing P05093 CP17A_HUMAN, 432 of 505\n",
      "Processing P02679 FIBG_HUMAN, 433 of 505\n",
      "Processing P09651 ROA1_HUMAN, 434 of 505\n",
      "Processing Q13485 SMAD4_HUMAN, 435 of 505\n",
      "Processing O60911 CATL2_HUMAN, 436 of 505\n",
      "Processing P11362 FGFR1_HUMAN, 437 of 505\n",
      "Processing P11717 MPRI_HUMAN, 438 of 505\n",
      "Processing Q16678 CP1B1_HUMAN, 439 of 505\n",
      "Processing Q9UM07 PADI4_HUMAN, 440 of 505\n",
      "Processing P25054 nan, 441 of 505\n",
      "Processing P16422 EPCAM_HUMAN, 442 of 505\n",
      "Processing Q16625 OCLN_HUMAN, 443 of 505\n",
      "Processing P15924 DESP_HUMAN, 444 of 505\n",
      "Processing P21912 DHSB_HUMAN, 445 of 505\n",
      "Processing P26038 MOES_HUMAN, 446 of 505\n",
      "Processing P27701 CD82_HUMAN, 447 of 505\n",
      "Processing P30043 BLVRB_HUMAN, 448 of 505\n",
      "Processing P40261 NNMT_HUMAN, 449 of 505\n",
      "Processing P67936 TPM4_HUMAN, 450 of 505\n",
      "Processing Q02413 DSG1_HUMAN, 451 of 505\n",
      "Processing Q15436 SC23A_HUMAN, 452 of 505\n",
      "Processing Q99527 GPER_HUMAN, 453 of 505\n",
      "Processing P26715 NKG2A_HUMAN, 454 of 505\n",
      "Processing O75795 UDB17_HUMAN, 455 of 505\n",
      "Processing P05120 PAI2_HUMAN, 456 of 505\n",
      "Processing P09486 SPRC_HUMAN, 457 of 505\n",
      "Processing P09544 WNT2_HUMAN, 458 of 505\n",
      "Processing P16112 PGCA_HUMAN, 459 of 505\n",
      "Processing P35613 BASI_HUMAN, 460 of 505\n",
      "Processing P41221 WNT5A_HUMAN, 461 of 505\n",
      "Processing Q07820 MCL1_HUMAN, 462 of 505\n",
      "Processing Q12964 Q12964_HUMAN, 463 of 505\n",
      "Gene 'MUC8' not found in the expression data.\n",
      "Processing Q13641 TPBG_HUMAN, 464 of 505\n",
      "Processing Q01650 LAT1_HUMAN, 465 of 505\n",
      "Processing Q03135 CAV1_HUMAN, 466 of 505\n",
      "Processing Q03167 TGBR3_HUMAN, 467 of 505\n",
      "Processing O95863 SNAI1_HUMAN, 468 of 505\n",
      "Processing O43623 SNAI2_HUMAN, 469 of 505\n",
      "Processing Q01826 SATB1_HUMAN, 470 of 505\n",
      "Processing Q9NYL2 MLTK_HUMAN, 471 of 505\n",
      "Gene 'MAP3K20' not found in the expression data.\n",
      "Processing Q15303 ERBB4_HUMAN, 472 of 505\n",
      "Processing P78545 ELF3_HUMAN, 473 of 505\n",
      "Processing Q6UXB2 VCC1_HUMAN, 474 of 505\n",
      "Processing P51692 STA5B_HUMAN, 475 of 505\n",
      "Processing P17252 KPCA_HUMAN, 476 of 505\n",
      "Processing P07288 KLK3_HUMAN, 477 of 505\n",
      "Processing P22352 GPX3_HUMAN, 478 of 505\n",
      "Processing P24462 CP3A7_HUMAN, 479 of 505\n",
      "Processing O00204 ST2B1_HUMAN, 480 of 505\n",
      "Processing P16662 UD2B7_HUMAN, 481 of 505\n",
      "Processing P09211 uniprot_accession\n",
      "P09211    GSTP1_HUMAN\n",
      "P09211    GSTP1_HUMAN\n",
      "Name: swiss_prot, dtype: object, 482 of 505\n",
      "Processing Q9UHF7 TRPS1_HUMAN, 483 of 505\n",
      "Processing P49765 VEGFB_HUMAN, 484 of 505\n",
      "Processing Q96D21 RHES_HUMAN, 485 of 505\n",
      "Processing Q8IUK5 PLDX1_HUMAN, 486 of 505\n",
      "Processing Q02224 CENPE_HUMAN, 487 of 505\n",
      "Processing P10275 ANDR_HUMAN, 488 of 505\n",
      "Processing O00744 WN10B_HUMAN, 489 of 505\n",
      "Processing Q9UMD9 COHA1_HUMAN, 490 of 505\n",
      "Processing Q02447 SP3_HUMAN, 491 of 505\n",
      "Processing P24592 IBP6_HUMAN, 492 of 505\n",
      "Processing P54852 EMP3_HUMAN, 493 of 505\n",
      "Processing P58335 ANTR2_HUMAN, 494 of 505\n",
      "Processing Q9GZU2 PEG3_HUMAN, 495 of 505\n",
      "Processing Q9UNQ0 ABCG2_HUMAN, 496 of 505\n",
      "Processing Q9Y243 AKT3_HUMAN, 497 of 505\n",
      "Processing Q96AM1 MRGRF_HUMAN, 498 of 505\n",
      "Processing P35749 MYH11_HUMAN, 499 of 505\n",
      "Processing Q9UMS6 SYNP2_HUMAN, 500 of 505\n",
      "Processing Q8N2R0 OSR2_HUMAN, 501 of 505\n",
      "Processing P29536 LMOD1_HUMAN, 502 of 505\n",
      "Processing O60894 RAMP1_HUMAN, 503 of 505\n",
      "Processing O43294 TGFI1_HUMAN, 504 of 505\n",
      "Processing Q13642 FHL1_HUMAN, 505 of 505\n"
     ]
    }
   ],
   "source": [
    "entry = 0\n",
    "for up_acc in data.index:\n",
    "    try:\n",
    "        entry += 1\n",
    "        print(f'Processing {up_acc} ') #{data.loc[up_acc, 'swiss_prot']}, {entry} of {len(data.index)}')\n",
    "\n",
    "        results_dir = f'/home/leandro/Insync/gdrive/Mimark/code/cohort-analysis/notebooks/other_analyses/antigens/data/results_screenEC/{up_acc[0:2]}/{up_acc}/'\n",
    "        FileHandler.ensureDir(results_dir)\n",
    "\n",
    "        # Step 1: Get the Uniprot object\n",
    "        step = 1\n",
    "        uniprot_object = OnlineOrf(up_acc)\n",
    "        step = 1.1\n",
    "        glycosilation_sites = uniprot_object.getGlycosylationSites()\n",
    "        if len(uniprot_object.getGlycosylationSites()) > 0:\n",
    "            glycosylation_df = pd.DataFrame(uniprot_object.getGlycosylationSites(), columns=['res', 'glycosylation'])\n",
    "            glycosylation_df.to_csv(results_dir + f'{up_acc}_glycosylation.csv')\n",
    "        \n",
    "        step = 1.2\n",
    "        modified_residues = uniprot_object.getModifiedResidues()\n",
    "        if len(uniprot_object.getModifiedResidues()) > 0:\n",
    "            modified_residues_df = pd.DataFrame(uniprot_object.getModifiedResidues(), columns=['res', 'modification'])\n",
    "            modified_residues_df.to_csv(results_dir + f'{up_acc}_modifications.csv')\n",
    "        \n",
    "        # step = 1.3\n",
    "        try:\n",
    "            isoforms = uniprot_object.getIsoforms()\n",
    "        except:\n",
    "            isoforms = []\n",
    "        step = 1.4\n",
    "        uniprot_string_id = uniprot_object.getStringId()\n",
    "        step = 1.5\n",
    "        try:\n",
    "            subcellular_location = uniprot_object.getSubcellularLocation()\n",
    "        except:\n",
    "            subcellular_location = None\n",
    "        step=1.6\n",
    "        crytals = uniprot_object.getCrystals()\n",
    "\n",
    "        # Step 2: Get the AlphaFold object\n",
    "        step = 2\n",
    "        alphafold_object = AlphaFoldHandler().get_model(up_acc)\n",
    "\n",
    "        # Step 3: Run discotope 3.0 and save the results, then load them\n",
    "        step = 3\n",
    "        if alphafold_object is not None:\n",
    "            discotope3_file = f'{up_acc}_A_discotope3.csv'\n",
    "            # If no results, run discotope\n",
    "            # if not FileHandler.fileExists(path=results_dir + discotope3_file):\n",
    "            #     pf.run_discotope(up_acc, save_to=results_dir)\n",
    "            \n",
    "            # Now, if the results are there, load them\n",
    "            if FileHandler.fileExists(path=results_dir + discotope3_file):\n",
    "                discotope_results = pd.read_csv(results_dir + discotope3_file, index_col=0)\n",
    "            else:\n",
    "                discotope_results = None\n",
    "\n",
    "        # Step 4: Compute the expression \n",
    "        step = 4\n",
    "        exp_file = f'{up_acc}_exp_results.csv'\n",
    "        #if FileHandler.fileExists(results_dir + exp_file):\n",
    "        #    exp_results = pd.read_csv(results_dir + exp_file, index_col=0)\n",
    "        #else:\n",
    "        try:\n",
    "            exp_auc, exp_normal, exp_cancer = pf.compare_gene_expression(gene_name=uniprot_object.getGeneName(), up_acc=up_acc, save_to=results_dir)\n",
    "        except:\n",
    "            exp_auc, exp_normal, exp_cancer = None, None, None\n",
    "        \n",
    "        exp_results = pd.DataFrame({'exp_auc': [exp_auc], 'exp_normal': [exp_normal], 'exp_cancer': [exp_cancer]})\n",
    "        exp_results.to_csv(results_dir + exp_file)\n",
    "\n",
    "        # Step 5: Align the isoforms\n",
    "        step = 5\n",
    "        if not FileHandler.fileExists(results_dir + f'{up_acc}_isoforms_alignment.html') and len(isoforms) > 1:\n",
    "            secuences = []\n",
    "            names = []\n",
    "            for isoform in isoforms:\n",
    "                isoform_obj = OnlineOrf(isoform)\n",
    "                secuences.append(isoform_obj.getSequence())\n",
    "                names.append(isoform)\n",
    "            alignment = align_sequences_biopython(secuences, names)\n",
    "            display_alignment_with_highlighting(alignment, save_to=results_dir + f'{up_acc}_isoforms_alignment.html')\n",
    "\n",
    "        # Step 6: Compute the aggregation propensity\n",
    "        step = 6\n",
    "        if alphafold_object is not None:\n",
    "            if not FileHandler.fileExists(results_dir + f'{up_acc}_agg.txt'):\n",
    "                TangoHandler.getAggregation(alphafold_object,results_dir)\n",
    "\n",
    "            agg_df = pd.read_csv(results_dir + f'{up_acc}_agg.txt', sep='\\t', header=0, index_col=\"res\")[[\"Aggregation\"]]\n",
    "        \n",
    "        # Step 7: Get interaction partners\n",
    "        step = 7\n",
    "        if not FileHandler.fileExists(results_dir + f'{up_acc}_interactors.tsv'):\n",
    "            interactors = pf.get_interactors(uniprot_string_id, up_acc, save_to=results_dir)\n",
    "        else:\n",
    "            interactors = pd.read_csv(results_dir + f'{up_acc}_interactors.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "        # Step 8: Get the protein homology\n",
    "        step = 8\n",
    "        if not FileHandler.fileExists(results_dir + f'{up_acc}_homologs.tsv'):\n",
    "            homology = pf.get_homologs(uniprot_object.getGeneName(), up_acc, save_to=results_dir)\n",
    "        else:\n",
    "            homology = pd.read_csv(results_dir + f'{up_acc}_homologs.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "        # Step 9: Get the protein structures bioassemblies to check for homo and multimers\n",
    "        step = 9\n",
    "        if not FileHandler.fileExists(results_dir + f'{up_acc}_bioassemblies.csv'):\n",
    "            try:\n",
    "                all_dataframes = []\n",
    "                for crystal in crytals:\n",
    "                    structure = Structure(crystal, replaceExistent=False)\n",
    "                    df = structure.analyzeStructureType()\n",
    "                    df['crystal_id'] = crystal\n",
    "                    all_dataframes.append(df)\n",
    "\n",
    "                if len(all_dataframes) > 1:\n",
    "                    bioassemblies_df = pd.concat(all_dataframes)\n",
    "                elif len(all_dataframes) == 1:\n",
    "                    bioassemblies_df = all_dataframes[0]\n",
    "                else:\n",
    "                    bioassemblies_df = None\n",
    "\n",
    "                if len(all_dataframes) > 0:\n",
    "                    bioassemblies_df.to_csv(results_dir+f'{up_acc}_bioassemblies.csv')\n",
    "            except:\n",
    "                bioassemblies_df = None\n",
    "        else:\n",
    "            bioassemblies_df = pd.read_csv(results_dir + f'{up_acc}_bioassemblies.csv')\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f'Error with {up_acc} in step {step}: {e}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Auxiliary: epitopes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "up_accs = {\n",
    "    \"KPYM\": \"P14618\",\n",
    "    \"MMP9\": \"P14780\",\n",
    "    \"HSPB1\": \"P04792\",\n",
    "    \"AGRIN\": \"O00468\",\n",
    "    \"CLIC1\": \"O00299\"\n",
    "}\n",
    "\n",
    "epitopes = {\n",
    "    \"KPYM\": [\"EAVRMQHLIARE\"],\n",
    "    \"MMP9\": [\"TFLGKEY\", \"GYPFDGKD\"],\n",
    "    \"HSPB1\": [\"KDGVV\"],\n",
    "    \"AGRIN\": [\"RLELSRHW\", \"FVGAGLRGC\", \"NPCHGAAPC\", \"RDRRLEF\", \"GHPCLNGASC\", \"VCLCPGGF\"],\n",
    "    \"CLIC1\": [\"KRRTE\", \"QVELF\", \"KRRTET\"] \n",
    "}\n",
    "\n",
    "\n",
    "up_objs = {k: OnlineOrf(v) for k, v in up_accs.items()}\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing KPYM\n",
      "Epitope: EAVRMQHLIARE\n",
      "Positions: [(372, 384)]\n",
      "\n",
      "\n",
      "\n",
      "Processing MMP9\n",
      "Epitope: TFLGKEY\n",
      "Positions: [(351, 358)]\n",
      "Epitope: GYPFDGKD\n",
      "Positions: [(177, 185)]\n",
      "\n",
      "\n",
      "\n",
      "Processing HSPB1\n",
      "Epitope: KDGVV\n",
      "Positions: [(113, 118)]\n",
      "\n",
      "\n",
      "\n",
      "Processing AGRIN\n",
      "Epitope: RLELSRHW\n",
      "Positions: [(1455, 1463)]\n",
      "Epitope: FVGAGLRGC\n",
      "Positions: [(1510, 1519)]\n",
      "Epitope: NPCHGAAPC\n",
      "Positions: [(1594, 1603)]\n",
      "Epitope: RDRRLEF\n",
      "Positions: [(1695, 1702)]\n",
      "Epitope: GHPCLNGASC\n",
      "Positions: [(1826, 1836)]\n",
      "Epitope: VCLCPGGF\n",
      "Positions: [(1843, 1851)]\n",
      "\n",
      "\n",
      "\n",
      "Processing CLIC1\n",
      "Epitope: KRRTE\n",
      "Positions: [(48, 53)]\n",
      "Epitope: QVELF\n",
      "Positions: [(6, 11)]\n",
      "Epitope: KRRTET\n",
      "Positions: [(48, 54)]\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "def find_all_epitopes(seq, epitope):\n",
    "    positions = []\n",
    "    for i in range(len(seq) - len(epitope)):\n",
    "        if seq[i:i+len(epitope)] == epitope:\n",
    "            positions.append((i, i+len(epitope)))\n",
    "    return positions\n",
    "\n",
    "for up in up_objs.keys():\n",
    "    print(f'Processing {up}')\n",
    "    up_obj = up_objs[up]\n",
    "    seq = up_obj.getSequence()\n",
    "\n",
    "    for epitope in epitopes[up]:\n",
    "        print(f'Epitope: {epitope}')\n",
    "        print(f'Positions: {find_all_epitopes(seq, epitope)}')\n",
    "\n",
    "    print('\\n\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
