# Detailed Data for P62851


## Introduction to the Detailed Summary

### How to Interpret the Results

- **Summary & Metrics**: This section provides a quick reference to essential protein attributes, including expression changes, family classification, and biomarker applications. Regulation status (upregulated/downregulated) indicates the protein's behavior in a disease context. Some information comes from the original excel file with the proteins selected from literature, while others are derived from the analyses.
- **Expression Comparison**: A visual representation comparing protein expression between normal and disease states. It highlights significant changes in expression levels that might indicate diagnostic or therapeutic relevance. This is data coming from transcriptomics experiments and could not translate similarly to protein levels.
- **Isoform Alignment**: An interactive view of isoform alignments, revealing structural and functional differences between variants of the protein.
- **Interactors & Homologs**: Tables listing known interaction partners and homologous proteins, the more interactors and homologs, the more complex the protein is to design an antibody for.
- **Biological Assemblies**: Information about the structural arrangement of the protein in different assemblies, providing insights into its functional state but also the complexity of the protein to develop antibodies.
- **Combined Per-Residue Information**: A detailed table summarizing residue-level data. This includes predictions for epitope regions, aggregation tendencies, and modifications that might impact the protein's function. Each row corresponds to a residue in the protein, providing insights into specific sites that may be important for research or drug development.
## Summary & Metrics

- **UniProt Accession**: P62851
- **Gene Name**: RPS25
- **Protein Name**: ribosomal protein S25
- **Swiss Prot**: nan
- **Family**: other
- **Biomarker Application**:  
- **Number of Isoforms**: 0
- **Regulation**: 1
- **(transcriptomics) AUC**: nan
- **(transcriptomics) Fold Change**: nan
- **(transcriptomics) Regulation**: Downregulated
- **Discotope Epitope Count**: 13
- **Max n_uniprots (Homo)**: N/A
- **Max n_uniprots (Hetero)**: 11.0


## Interactors

| preferredName_A   | preferredName_B   |   score |
|:------------------|:------------------|--------:|
| RPS25             | RPL26             |   0.999 |
| RPS25             | UBA52             |   0.999 |
| RPS25             | RPS6              |   0.999 |
| RPS25             | RPS15             |   0.999 |
| RPS25             | RPS2              |   0.999 |
| RPS25             | RPS8              |   0.999 |
| RPS25             | RPL37A            |   0.999 |
| RPS25             | RPL24             |   0.999 |
| RPS25             | RPS20             |   0.999 |
| RPS25             | RPL8              |   0.999 |
| RPS25             | RPL17             |   0.999 |
| RPS25             | RPL23A            |   0.999 |
| RPS25             | RPL10A            |   0.999 |
| RPS25             | RPS5              |   0.999 |
| RPS25             | RPS28             |   0.999 |
| RPS25             | RPS4X             |   0.999 |
| RPS25             | RPS7              |   0.999 |
| RPS25             | RPL4              |   0.999 |
| RPS25             | RPLP1             |   0.999 |
| RPS25             | RPS17             |   0.999 |
| RPS25             | RPL30             |   0.999 |
| RPS25             | RPL5              |   0.999 |
| RPS25             | RPL19             |   0.999 |
| RPS25             | RPS16             |   0.999 |
| RPS25             | RPS12             |   0.999 |
| RPS25             | RPL10             |   0.999 |
| RPS25             | RPS18             |   0.999 |
| RPS25             | RPL34             |   0.999 |
| RPS25             | RPSA              |   0.999 |
| RPS25             | RPL35             |   0.999 |
| RPS25             | RACK1             |   0.999 |
| RPS25             | RPL15             |   0.999 |
| RPS25             | RPS19             |   0.999 |
| RPS25             | RPL23             |   0.999 |
| RPS25             | RPL36             |   0.999 |
| RPS25             | RPL6              |   0.999 |
| RPS25             | RPL7A             |   0.999 |
| RPS25             | RPL27A            |   0.999 |
| RPS25             | RPS10             |   0.999 |
| RPS25             | RPL13             |   0.999 |
| RPS25             | RPL9              |   0.999 |
| RPS25             | RPL18             |   0.999 |
| RPS25             | RPS21             |   0.999 |
| RPS25             | RPL7              |   0.999 |
| RPS25             | RPL11             |   0.999 |
| RPS25             | RPS24             |   0.999 |
| RPS25             | RPS3              |   0.999 |
| RPS25             | RPS27A            |   0.999 |
| RPS25             | RPS13             |   0.999 |
| RPS25             | RPS3A             |   0.999 |

## Homologs

| uniprot_id   | gene_id   |
|:-------------|:----------|
| P62851       | RPS25     |

## Biological Assemblies

|   Unnamed: 0 | crystal_id   |   assembly |   n_uniprots | composition   |
|-------------:|:-------------|-----------:|-------------:|:--------------|
|            0 | 5oa3         |          1 |           11 | Hetero        |
|            0 | 7wtt         |          1 |           11 | Hetero        |
|            0 | 6zxf         |          1 |           11 | Hetero        |
|            0 | 6zmt         |          1 |           11 | Hetero        |
|            0 | 7wtv         |          1 |           11 | Hetero        |
|            0 | 7wtw         |          1 |           11 | Hetero        |
|            0 | 8t4s         |          1 |           11 | Hetero        |
|            0 | 6g5h         |          1 |           11 | Hetero        |
|            0 | 6zlw         |          1 |           11 | Hetero        |
|            0 | 7wtx         |          1 |           11 | Hetero        |
|            0 | 6zxe         |          1 |           11 | Hetero        |
|            0 | 6g4w         |          1 |           11 | Hetero        |
|            0 | 6zv6         |          1 |           11 | Hetero        |
|            0 | 7wu0         |          1 |           11 | Hetero        |
|            0 | 6zuo         |          1 |           11 | Hetero        |
|            0 | 7wtu         |          1 |           11 | Hetero        |
|            0 | 6g5i         |          1 |           11 | Hetero        |
|            0 | 6g18         |          1 |           11 | Hetero        |
|            0 | 6ybs         |          1 |           11 | Hetero        |
|            0 | 7tql         |          1 |           11 | Hetero        |
|            0 | 6zxh         |          1 |           11 | Hetero        |
|            0 | 7r4x         |          1 |           11 | Hetero        |
|            0 | 5flx         |          1 |           11 | Hetero        |
|            0 | 8ppk         |          1 |           11 | Hetero        |
|            0 | 6zvh         |          1 |           11 | Hetero        |
|            0 | 6g53         |          1 |           11 | Hetero        |
|            0 | 6g4s         |          1 |           11 | Hetero        |
|            0 | 6zn5         |          1 |           11 | Hetero        |
|            0 | 6g51         |          1 |           11 | Hetero        |
|            0 | 6zoj         |          1 |           11 | Hetero        |
|            0 | 6zxg         |          1 |           11 | Hetero        |
|            0 | 7k5i         |          1 |           11 | Hetero        |
|            0 | 6zol         |          1 |           11 | Hetero        |
|            0 | 7wtz         |          1 |           11 | Hetero        |
|            0 | 6zxd         |          1 |           11 | Hetero        |
|            0 | 5a2q         |          1 |           11 | Hetero        |

## Combined Per-Residue Information

|   res | aa   |   epitope_score | epitope   |   relative_surface_accessibility |   modeling_confidence |   Aggregation | modification                 |
|------:|:-----|----------------:|:----------|---------------------------------:|----------------------:|--------------:|:-----------------------------|
|     1 | M    |         0.25552 | False     |                          1.28019 |                 43.38 |         0     | N/A                          |
|     2 | P    |         0.23123 | False     |                          0.92821 |                 53.45 |         0     | N/A                          |
|     3 | P    |         0.23246 | False     |                          0.96781 |                 72.73 |         0     | N/A                          |
|     4 | K    |         0.22952 | False     |                          0.93154 |                 54.2  |         0     | N/A                          |
|     5 | D    |         0.30397 | False     |                          0.92695 |                 65.57 |         0     | N/A                          |
|     6 | D    |         0.23813 | False     |                          0.75342 |                 58.9  |         0     | N/A                          |
|     7 | K    |         0.251   | False     |                          0.90473 |                 48.91 |         0     | N/A                          |
|     8 | K    |         0.22655 | False     |                          0.9992  |                 44.43 |         0     | N/A                          |
|     9 | K    |         0.23643 | False     |                          0.98813 |                 42.88 |         0     | N/A                          |
|    10 | K    |         0.32206 | False     |                          0.89373 |                 41.34 |         0     | N/A                          |
|    11 | D    |         0.28042 | False     |                          0.78762 |                 40.61 |         0     | N/A                          |
|    12 | A    |         0.2619  | False     |                          0.86528 |                 41.82 |         0     | N/A                          |
|    13 | G    |         0.30962 | False     |                          0.95496 |                 41.78 |         0     | N/A                          |
|    14 | K    |         0.29357 | False     |                          1.01982 |                 39.89 |         0     | N/A                          |
|    15 | S    |         0.24497 | False     |                          0.80579 |                 39.32 |         0     | N/A                          |
|    16 | A    |         0.3307  | False     |                          0.97716 |                 44.86 |         0     | N/A                          |
|    17 | K    |         0.33158 | False     |                          0.97916 |                 39.69 |         0     | N/A                          |
|    18 | K    |         0.35123 | False     |                          0.9847  |                 40.32 |         0     | N/A                          |
|    19 | D    |         0.3089  | False     |                          0.80063 |                 41.78 |         0     | N/A                          |
|    20 | K    |         0.28329 | False     |                          0.9534  |                 43.08 |         0     | N/A                          |
|    21 | D    |         0.24565 | False     |                          0.89665 |                 40.71 |         0     | N/A                          |
|    22 | P    |         0.28091 | False     |                          0.90196 |                 49.04 |         0     | N/A                          |
|    23 | V    |         0.33598 | False     |                          0.97729 |                 40.36 |         0     | N/A                          |
|    24 | N    |         0.26879 | False     |                          0.83292 |                 40.58 |         0     | N/A                          |
|    25 | K    |         0.3529  | False     |                          0.99008 |                 43.7  |         0     | N/A                          |
|    26 | S    |         0.27915 | False     |                          0.87825 |                 41.2  |         0     | N/A                          |
|    27 | G    |         0.27459 | False     |                          0.8575  |                 47.78 |         0     | N/A                          |
|    28 | G    |         0.36764 | False     |                          0.81802 |                 46.48 |         0     | N/A                          |
|    29 | K    |         0.37619 | False     |                          1.02276 |                 49.09 |         0     | N/A                          |
|    30 | A    |         0.32532 | False     |                          0.91034 |                 58.52 |         0     | N/A                          |
|    31 | K    |         0.2555  | False     |                          1.01269 |                 64.2  |         0     | N/A                          |
|    32 | K    |         0.27929 | False     |                          0.91897 |                 73.54 |         0     | N/A                          |
|    33 | K    |         0.20994 | False     |                          0.75019 |                 75.19 |         0     | N/A                          |
|    34 | K    |         0.27326 | False     |                          0.97054 |                 74.52 |         0     | N/A                          |
|    35 | W    |         0.27892 | False     |                          0.91439 |                 66.89 |         0     | N/A                          |
|    36 | S    |         0.29615 | False     |                          0.73298 |                 71.39 |         0     | N/A                          |
|    37 | K    |         0.31638 | False     |                          1.0422  |                 63.18 |         0     | N/A                          |
|    38 | G    |         0.36211 | False     |                          0.8533  |                 69.33 |         0     | N/A                          |
|    39 | K    |         0.37125 | False     |                          0.97892 |                 69.36 |         0     | N/A                          |
|    40 | V    |         0.29059 | False     |                          1.01346 |                 76.56 |         0     | N/A                          |
|    41 | R    |         0.35167 | False     |                          0.8305  |                 77.16 |         0     | N/A                          |
|    42 | D    |         0.23307 | False     |                          0.72588 |                 84.22 |         0     | N/A                          |
|    43 | K    |         0.27886 | False     |                          0.88732 |                 83.15 |         0     | N6-acetyllysine              |
|    44 | L    |         0.43486 | True      |                          0.8359  |                 80.72 |        10.853 | N/A                          |
|    45 | N    |         0.36322 | False     |                          0.79271 |                 86.31 |        11.118 | N/A                          |
|    46 | N    |         0.44207 | True      |                          0.31446 |                 88.6  |        29.725 | N/A                          |
|    47 | L    |         0.36336 | False     |                          0.413   |                 87.81 |        31.566 | N/A                          |
|    48 | V    |         0.38269 | False     |                          0.79204 |                 87.04 |        31.807 | N/A                          |
|    49 | L    |         0.37676 | False     |                          0.82419 |                 88.54 |        31.807 | N/A                          |
|    50 | F    |         0.29741 | False     |                          0.25314 |                 87.02 |        31.69  | N/A                          |
|    51 | D    |         0.30808 | False     |                          0.43374 |                 88.53 |         2.082 | N/A                          |
|    52 | K    |         0.30375 | False     |                          0.8569  |                 88.64 |         2.082 | N6-acetyllysine; alternate   |
|    52 | K    |         0.30375 | False     |                          0.8569  |                 88.64 |         2.082 | N6-succinyllysine; alternate |
|    53 | A    |         0.2484  | False     |                          0.59622 |                 91.01 |         1.798 | N/A                          |
|    54 | T    |         0.21206 | False     |                          0.12853 |                 88.47 |         1.56  | N/A                          |
|    55 | Y    |         0.29806 | False     |                          0.33852 |                 89.59 |         1.56  | N/A                          |
|    56 | D    |         0.15438 | False     |                          0.38987 |                 89.89 |         0     | N/A                          |
|    57 | K    |         0.25518 | False     |                          0.52139 |                 89.31 |         0     | N/A                          |
|    58 | L    |         0.03662 | False     |                          0.01649 |                 89.18 |         0     | N/A                          |
|    59 | C    |         0.17011 | False     |                          0.29621 |                 87.33 |         0     | N/A                          |
|    60 | K    |         0.32891 | False     |                          0.73652 |                 87.01 |         0     | N6-acetyllysine              |
|    61 | E    |         0.41137 | True      |                          0.35058 |                 87.83 |         0     | N/A                          |
|    62 | V    |         0.00656 | False     |                          0.00095 |                 87.84 |         0     | N/A                          |
|    63 | P    |         0.20943 | False     |                          0.17682 |                 88.37 |         0     | N/A                          |
|    64 | N    |         0.23022 | False     |                          0.68965 |                 88.36 |         0     | N/A                          |
|    65 | Y    |         0.23849 | False     |                          0.18192 |                 88.83 |         0     | N/A                          |
|    66 | K    |         0.30524 | False     |                          0.83196 |                 89.82 |         0     | N6-acetyllysine              |
|    67 | L    |         0.28812 | False     |                          0.61398 |                 91.13 |         0     | N/A                          |
|    68 | I    |         0.00609 | False     |                          0       |                 92.22 |         0     | N/A                          |
|    69 | T    |         0.31318 | False     |                          0.09552 |                 93.12 |         0     | N/A                          |
|    70 | P    |         0.13208 | False     |                          0.15283 |                 92.28 |         0     | N/A                          |
|    71 | A    |         0.27672 | False     |                          0.66214 |                 92.45 |         0     | N/A                          |
|    72 | V    |         0.26291 | False     |                          0.23108 |                 92.32 |         0     | N/A                          |
|    73 | V    |         0.00441 | False     |                          0.00286 |                 91.38 |         0     | N/A                          |
|    74 | S    |         0.2277  | False     |                          0.18111 |                 90.58 |         0     | N/A                          |
|    75 | E    |         0.40177 | False     |                          0.72876 |                 89.82 |         0     | N/A                          |
|    76 | R    |         0.39031 | False     |                          0.58139 |                 89.28 |         0     | N/A                          |
|    77 | L    |         0.11501 | False     |                          0.07836 |                 88.52 |         0     | N/A                          |
|    78 | K    |         0.36057 | False     |                          0.68232 |                 89.46 |         0     | N/A                          |
|    79 | I    |         0.01339 | False     |                          0       |                 91.33 |         0     | N/A                          |
|    80 | R    |         0.41843 | True      |                          0.54018 |                 92.26 |         0     | N/A                          |
|    81 | G    |         0.30546 | False     |                          0.34754 |                 91.48 |         0     | N/A                          |
|    82 | S    |         0.39565 | False     |                          0.58019 |                 90.93 |         0     | N/A                          |
|    83 | L    |         0.44132 | True      |                          0.22175 |                 91.62 |         0     | N/A                          |
|    84 | A    |         0.0048  | False     |                          0       |                 92.77 |         0     | N/A                          |
|    85 | R    |         0.42267 | True      |                          0.57145 |                 91.92 |         0     | N/A                          |
|    86 | A    |         0.41328 | True      |                          0.43197 |                 92.03 |         0     | N/A                          |
|    87 | A    |         0.052   | False     |                          0.0175  |                 92.33 |         0     | N/A                          |
|    88 | L    |         0.0114  | False     |                          0.0033  |                 91.88 |         0     | N/A                          |
|    89 | Q    |         0.37912 | False     |                          0.55997 |                 91.25 |         0     | N/A                          |
|    90 | E    |         0.48772 | True      |                          0.43642 |                 90.83 |         0     | N/A                          |
|    91 | L    |         0.03708 | False     |                          0.00989 |                 89.97 |         0     | N/A                          |
|    92 | L    |         0.24663 | False     |                          0.38906 |                 88.77 |         0     | N/A                          |
|    93 | S    |         0.27383 | False     |                          0.79372 |                 89.01 |         0     | N/A                          |
|    94 | K    |         0.40425 | False     |                          0.55908 |                 88.43 |         0     | N6-succinyllysine; alternate |
|    94 | K    |         0.40425 | False     |                          0.55908 |                 88.43 |         0     | N6-acetyllysine; alternate   |
|    95 | G    |         0.09733 | False     |                          0.43969 |                 85.31 |         0     | N/A                          |
|    96 | L    |         0.33926 | False     |                          0.42458 |                 86.63 |         0     | N/A                          |
|    97 | I    |         0.08308 | False     |                          0.03497 |                 90.33 |         0     | N/A                          |
|    98 | K    |         0.26931 | False     |                          0.63211 |                 89.64 |         0     | N/A                          |
|    99 | L    |         0.41237 | True      |                          0.3788  |                 91.16 |         0     | N/A                          |
|   100 | V    |         0.29676 | False     |                          0.63372 |                 89.84 |         0     | N/A                          |
|   101 | S    |         0.42916 | True      |                          0.40887 |                 89.38 |         0     | N/A                          |
|   102 | K    |         0.40664 | False     |                          0.80422 |                 89.82 |         0     | N/A                          |
|   103 | H    |         0.43829 | True      |                          0.62492 |                 91.22 |         0     | N/A                          |
|   104 | R    |         0.43869 | True      |                          0.97015 |                 87.16 |         0     | N/A                          |
|   105 | A    |         0.39176 | False     |                          0.77359 |                 84.95 |         0     | N/A                          |
|   106 | Q    |         0.3738  | False     |                          0.51357 |                 86.95 |         0     | N/A                          |
|   107 | V    |         0.20891 | False     |                          0.2296  |                 90.65 |         0     | N/A                          |
|   108 | I    |         0.30911 | False     |                          0.26879 |                 93.04 |         0     | N/A                          |
|   109 | Y    |         0.11532 | False     |                          0.04112 |                 92.54 |         0     | N/A                          |
|   110 | T    |         0.23047 | False     |                          0.23428 |                 91.52 |         0     | N/A                          |
|   111 | R    |         0.2013  | False     |                          0.44439 |                 89.5  |         0     | N/A                          |
|   112 | N    |         0.16335 | False     |                          0.39568 |                 83.35 |         0     | N/A                          |
|   113 | T    |         0.44576 | True      |                          0.79702 |                 75.92 |         0     | N/A                          |
|   114 | K    |         0.28922 | False     |                          0.79465 |                 65.82 |         0     | N/A                          |
|   115 | G    |         0.39067 | False     |                          0.81952 |                 52.07 |         0     | N/A                          |
|   116 | G    |         0.32983 | False     |                          1.05924 |                 49.21 |         0     | N/A                          |
|   117 | D    |         0.29998 | False     |                          0.78818 |                 40.3  |         0     | N/A                          |
|   118 | A    |         0.22966 | False     |                          0.80448 |                 40.7  |         0     | N/A                          |
|   119 | P    |         0.25154 | False     |                          0.86732 |                 39.89 |         0     | N/A                          |
|   120 | A    |         0.24581 | False     |                          0.97115 |                 35.8  |         0     | N/A                          |
|   121 | A    |         0.23136 | False     |                          1.06749 |                 39.07 |         0     | N/A                          |
|   122 | G    |         0.31015 | False     |                          0.93849 |                 39.62 |         0     | N/A                          |
|   123 | E    |         0.23526 | False     |                          0.92798 |                 34.6  |         0     | N/A                          |
|   124 | D    |         0.20309 | False     |                          0.86732 |                 38.81 |         0     | N/A                          |
|   125 | A    |         0.17466 | False     |                          1.61093 |                 38.27 |         0     | N/A                          |

